src/DataObject/DataObjectDocument.php

```diff
@@ -642,7 +693,7 @@ public function onAddToSearchIndexes(string $event): void
 
                 $currentDataObject = $this->getDataObject();
 
-                $liveDataObject = DataObject::get($currentDataObject->ClassName)->byID($currentDataObject->ID);
+                $liveDataObject = $currentDataObject ? DataObject::get($currentDataObject->ClassName)->byID($currentDataObject->ID) : null;
```

Comment:
These changes are to do with the failing test on the backport branch (see https://github.com/silverstripe/silverstripe-search-service/actions/runs/8748084849/job/24007554595?pr=93) where a data object hasn't been set, and probably more to do with `testEvents` unit test not setting the data object on the class.

With regards to this change, if `$currentDataObject` is null, then this sets `$liveDataObject` to null and the error is thrown. This seems to be doing as intended (and not setting draft objects onto $liveDataObject) but i'll add some further tests so that we are testing this code for unversioned, draft and published and no dataobject set scenarios. Also, perhaps in the case of no dataobject (though this is probably unlikely when adding documents) the code should throw a different error to avoid confusion:

```
$currentDataObject = $this->getDataObject();

if (!$currentDataObject) {
    throw new IndexingServiceException('No data object set for indexing');
}

$liveDataObject = DataObject::get($currentDataObject->ClassName)->byID($currentDataObject->ID);

if (!$liveDataObject) {
    // unlikely case as indexer calls 'shouldIndex' immediately prior to this
    throw new IndexingServiceException('Only published DataObjects may be added to the index');
}

$this->setDataObject($liveDataObject);
```

Reply:
hmm yeah I didn't bring the tests over as it was a bit of effort to do so for the older branch. Ah yes looks like I read it too quickly and its not what I thought it was :) 

Are you ok using this fork for now on your projects? I'll probably move these changes in to the latest v3 branch for future support

Reply:
in other words - will you be going to CMS5 soon?